cmake/target/check.cmake

```diff
@@ -59,12 +61,17 @@ function(check_add_module_target MODULE_NAME TARGET_NAME SOURCE_FILES DEPENDENCI
     # Protects against multiple calls to fprime_register_ut()
     if (NOT BUILD_TESTING OR NOT MODULE_TYPE STREQUAL "Unit Test")
         return()
-    elseif (NOT TARGET ${MODULE_NAME}_${TARGET_NAME})
+    endif()
+    # UTs MODULE_NAME defaults to <FPRIME_MODULE_NAME>_ut_exe
+    # The below handling gives CHECK_TARGET_NAME = <FPRIME_MODULE_NAME>_check
+    string(REGEX REPLACE "_${UT_TARGET}$" "" CHECK_TARGET_NAME "${MODULE_NAME}")
+    string(APPEND CHECK_TARGET_NAME "_${TARGET_NAME}")
```

Comment:
Doing regex replace _then_ append so that it also works with UTs that register themselves with a hand-coded name, e.g. https://github.com/nasa/fprime/blob/95c42a0496ad5f0172da28ee23d0a962a0f3ff7d/Drv/Ip/CMakeLists.txt#L44